aspnetcore/fundamentals/portable-object-localization.md

```diff
@@ -0,0 +1,218 @@
+
+The Orchard Localization package provides an API to invoke these different plural forms automatically.
+
+### Creating pluralization PO files
+
+Add the following content to the previously mentioned *fr.po* file:
+
+```text
+msgid "There is one item."
+msgid_plural "There are {0} items."
+msgstr[0] "Il y a un élément."
+msgstr[1] "Il y a {0} éléments."
+```
+
+### Adding a language using different pluralization forms
+
+English and French strings were used in the previous example. English and French have only two pluralization forms and share the same form rules, which is that
+a cardinality of one is mapped to the first plural form. Any other cardinality is mapped to the second plural form.
+
+Unfortunately, not all languages share the same rules. This is illustrated with the Czech language, which has three plural forms.
+
+Create the `cs.po` file as follows, and note how the pluralization needs three different translations:
+
+[!code-text[Main](localization/sample/POLocalization/cs.po)]
+
+To accept Czech localizations, add `"cs"` to the list of supported cultures in the `ConfigureServices` method of *Startup.cs*:
+
+```csharp
+var supportedCultures = new List<CultureInfo>
+{
+    new CultureInfo("en-US"),
+    new CultureInfo("en"),
+    new CultureInfo("fr-FR"),
+    new CultureInfo("fr"),
+    new CultureInfo("cs")
+};
+```
+
+Edit the *Views/Home/About.cshtml* file to render localized, plural strings for several cardinalities:
+
+```cshtml
+<p>@Localizer.Plural(1, "There is one item.", "There are {0} items.")</p>
+<p>@Localizer.Plural(2, "There is one item.", "There are {0} items.")</p>
+<p>@Localizer.Plural(5, "There is one item.", "There are {0} items.")</p>
+```
+
+**Note:** In a real world scenario, a variable would be used to represent the count. Here, we repeat the same code with three different values to expose a very specific case.
+
+Upon switching cultures, you see the following:
+
+For `/Home/About`:
+
+```html
+There is one item.
+There are 2 items.
+There are 5 items.
+```
+
+For `/Home/About?culture=fr`:
+
+```html
+Il y a un élément.
+Il y a 2 éléments.
+Il y a 5 éléments.
+```
+
+For `/Home/About?culture=cs`:
+
+```html
+Existuje jedna položka.
+Existují 2 položky.
+Existuje 5 položek.
+```
+
+Note that for the Czech culture, the three translations are different. The French and English cultures share the same construction for the two last translated strings.
+
+## Advanced tasks
+
+### Contextualizing strings
+
+Applications often contain the same strings to be translated in several places, while requiring the flexibility to define different translations. A PO file supports the notion of a *context*, which can be used to categorize the string being represented.
```

Comment:
How about 

Applications often contain the ~same~ strings to be translated in several places, with the same string having a different translation in some places (views or using class files).

 A PO file supports the notion of a *file context*, which can be used to categorize the string being represented. Using a file context, a string can be translated differently, depending on the file context (or lack of a file context).